src/main/java/qna/domain/Question.java

```diff
@@ -86,32 +96,37 @@ public boolean isDeleted() {
     public void setDeleted(boolean deleted) {
         this.deleted = deleted;
     }
-    @Override
-    public String toString() {
-        return "Question{" +
-                "id=" + id +
-                ", title='" + title + '\'' +
-                ", contents='" + contents + '\'' +
-                ", writer=" + writer +
-                ", deleted=" + deleted +
-                '}';
-    }
 
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         Question question = (Question) o;
-        return deleted == question.deleted &&
-                Objects.equals(id, question.id) &&
-                Objects.equals(title, question.title) &&
-                Objects.equals(contents, question.contents) &&
-                Objects.equals(writer, question.writer);
+        return Objects.equals(id, question.id);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(id, title, contents, writer, deleted);
+        return Objects.hash(id);
+    }
+
+    public boolean hasAnswers() {
+        return answers.hasAnswers();
+    }
+
+    private DeleteHistory createDeleteHistory() {
+        setDeleted(true);
+        return new DeleteHistory(ContentType.QUESTION, id, writer, LocalDateTime.now());
+    }
+
+    public DeleteHistories delete(User loginUser) throws CannotDeleteException {
```

Comment:
Q. 그리고 일급컬렉션 사용을 하면 getter를 사용해야만 레파지토리로 데이터를 전달할 수 있는데요. 이렇게 사용해도 괜찮은지 질문 남깁니다.

ex) DeleteHistories 내 컬렉션을 사용하기 위해 getter메소드를 사용하였습니다.

A. 넵 괜찮습니다~ 
외부에서 getter를 사용하는 것이 어색하다면 

```suggestion
    public List<DeleteHistory> delete(User loginUser) throws CannotDeleteException {
```

와 같이 넘겨주는 방법도 있을 것 같습니다 ㅎㅎ

Reply:
헛 이걸 빼먹다니.. 답변해주셔서 감사합니다 😄 😄 😄